pyiceberg/catalog/glue.py

```diff
@@ -117,6 +124,12 @@
 ICEBERG_FIELD_OPTIONAL = "iceberg.field.optional"
 ICEBERG_FIELD_CURRENT = "iceberg.field.current"
 
+GLUE_PROFILE_NAME = "glue.profile-name"
+GLUE_REGION = "glue.region"
+GLUE_ACCESS_KEY_ID = "glue.access-key-id"
```

Comment:
How common is it to have a separate `access-key-id` for glue and the table itself? The S3FileIO would not pick up the `glue.access-key-id`:

https://github.com/apache/iceberg-python/blob/3f44dfe711e96beda6aa8622cf5b0baffa6eb0f2/pyiceberg/io/pyarrow.py#L349

This way you would need to set both `glue.access-key-id` (which is used for authenticating against Glue), and then you need to set `s3.access-key-id` to ensure that it can access the metadata on the bucket. Maybe good to highlight this on the docs, WDYT?

I'm not an AWS expert, but my gut feeling is that normally people rely on `AWS_ACCESS_KEY_ID` to be picked up for both Glue and S3, but I prefer to be able to put this in the `~/.pyiceberg.yaml` as well.

Reply:
Thanks for the suggestions! I've updated the doc to explicitly indicating that `glue.*` properties are for Glue Catalog only. I've also added the `s3.*` properties to the example to make this clear.

>  but my gut feeling is that normally people rely on AWS_ACCESS_KEY_ID to be picked up for both Glue and S3, but I prefer to be able to put this in the ~/.pyiceberg.yaml as well.

The `client.*` properties will cover this case. When user set `client.*` properties, both catalog and FileIO can pick up the credentials. However, due to the limitation of pyarrow's S3FileSystem as noted in https://github.com/apache/iceberg-python/issues/570, we cannot yet support a unified `profile-name` property.

I added a separate section `Unified AWS Credentials` for the `client.*` properties. Please let me know WDYT.

Here are some preview of the updated page:
![Screenshot 2024-07-15 at 00 04 10](https://github.com/user-attachments/assets/6c32dd75-edc9-41ef-8e71-4a1e6f76630c)
![Screenshot 2024-07-15 at 00 05 18](https://github.com/user-attachments/assets/7964a762-7aec-4c09-9ad9-9b62f92bd0ee)

Reply:
That makes sense, I'm good with that now. At some point we should have a bigger conversation across languages to unify this.